### PR TITLE
Httperrorfixex

### PR DESCRIPTION
**Pull request by guzzijones**
**By submitting this pull-request, I confirm the following:**

- I have read and understood the [Contributing Guide](https://github.com/monero-project/kovri/blob/master/doc/CONTRIBUTING.md).
- I have checked that another pull-request for this purpose does not exist.
- I have considered and confirmed that this submission will be valuable to others.
- I accept that this submission may not be used and that this pull-request may be closed by the will of the maintainer.
- I give this submission freely under the BSD 3-clause license.

*Place an X inside the bracket (or click the box in preview) to confirm*
- [x ] I confirm.

---

fixes return response http body and uses appropriate codes for response.
A more complete fix is in the works on my local repo.

Sorry for the horrible title.  It used the name of my branch. I didn't change it.

### CODE REVIEW

File: src/client/proxy/http.h [hunk 1]
```diff
@@ -103,7 +104,108 @@ class HTTPProxyHandler
...
+    not_acceptable = 406,
+    request_timeout = 408,
+    precondition_failed = 412,
+    unsatisfiable_range = 416,
+    internal_server_error = 500,
+    not_implemented = 501,
+    bad_gateway = 502,
+    service_unavailable = 503,
+    space_unavailable = 507
+  };
```
Comment (anonimal) [inline src/client/proxy/http.h]: Please add a note that this was copypasta'd from cpp-netlib and a **TODO** that we should ultimately use cpp-netlib as a proxy (whenever someone gets to writing one).

Comment (anonimal) [inline src/client/proxy/http.h]: ^ this applies to all the member code, above and below.

File: src/client/proxy/http.cc [hunk 2]
```diff
@@ -321,20 +322,35 @@ void HTTPProxyHandler::HandleJumpService() {
...
+    std::string response =
+    "HTTP/1.0 " + std::to_string(statusCode) + " " +
+    status_message(statusCode)+"\r\n" +
+    "Content-type: text/html;charset=UTF-8\r\n" +
+    "Content-Encoding: UTF-8\r\n" +
+    "Content-length:" + std::to_string(htmlbody.size()) + "\r\n\r\n" + htmlbody;
+
   boost::asio::async_write(
-      *m_Socket,
+    *m_Socket,
```
Comment (anonimal) [inline src/client/proxy/http.cc]: ?

**Comment by anonimal** [comment]
Once resolved, closes #11.

File: src/client/proxy/http.cc (hunk 2 above)
Comment (guzzijones) [inline src/client/proxy/http.cc]: not sure here. i will revert that change.

File: src/client/proxy/http.h [hunk 3]
```diff
@@ -59,7 +59,8 @@ class HTTPProxyServer
       const std::string& name,
       const std::string& address,
       std::uint16_t port,
-      std::shared_ptr<kovri::client::ClientDestination> local_destination = nullptr);
+      std::shared_ptr<kovri::client::ClientDestination>
+        local_destination = nullptr);
```
Comment (guzzijones) [inline src/client/proxy/http.h]: oh i think i was going through guessing and changed this.

**Pull request merged**